File: backend/src/main/kotlin/com/tonihacks/annahwi/entity/TextVersion.kt
package com.tonihacks.annahwi.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "text_versions")
class TextVersion : PanacheEntityBase {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id", updatable = false, nullable = false)
    var id: UUID? = null
    
    @Column(name = "text_id", nullable = false)
    lateinit var textId: UUID
    
    @Column(name = "version_number", nullable = false)
    var versionNumber: Int = 1
    
    @Column(name = "title", length = 255, nullable = false)
    lateinit var title: String
    
    @Column(name = "arabic_content", columnDefinition = "TEXT", nullable = false)
    lateinit var arabicContent: String
    
    @Column(name = "transliteration", columnDefinition = "TEXT")
    var transliteration: String? = null
    
    @Column(name = "translation", columnDefinition = "TEXT")
    var translation: String? = null
    
    @Column(name = "comments", columnDefinition = "TEXT")
    var comments: String? = null
    
    @Column(name = "created_at", nullable = false)
    var createdAt: LocalDateTime = LocalDateTime.now()
    
    @Column(name = "updated_at", nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now()
    
    @PrePersist
    fun prePersist() {
        createdAt = LocalDateTime.now()
        updatedAt = LocalDateTime.now()
    }
    
    @PreUpdate
    fun preUpdate() {
        updatedAt = LocalDateTime.now()
    }
}

File: frontend/types/textVersion.ts
/**
 * Represents a summary of a text version, used in dropdowns
 */
export interface TextVersionSummary {
  id: string
  versionNumber: number
  updatedAt: string
  isCurrent: boolean
}

/**
 * Represents a text version with content
 */
export interface TextVersion extends TextVersionSummary {
  textId: string
  content: {
    title: string
    arabicContent: string
    transliteration?: string
    translation?: string
    comments?: string
  }
  createdAt: string
}

/**
 * Response from the API when fetching text versions
 */
export interface TextVersionsResponse {
  versions: TextVersionSummary[]
  currentVersionId: string
}
